src/hooks/useObserver.tsx

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from 'react';
+
+// Move types
+type useObserverHookData = { [key: string]: IntersectionObserverEntry };
+interface useObserverHook {
+  (
+    ref: React.MutableRefObject<HTMLElement[]>,
+    observerOptions: IntersectionObserverInit
+  ): useObserverHookData;
+}
+
+const useObserver: useObserverHook = (ref, observerOptions) => {
```

Comment:
A comment on what this does could be helpful :smile:

Reply:
✅